### PR TITLE
HPCC-28717 Allow planes and aliases to be restricted to certain components

### PR DESCRIPTION
**Pull request by ghalliday**
…nents

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith I think this is everything that is needed.  I feel sure I've missed something though!

File: helm/hpcc/templates/_helpers.tpl [hunk 1]
```diff
@@ -355,7 +356,8 @@ to addVolumeMounts so that if a plane can be used for multiple purposes then dup
    {{- /*Generate entries for each alias of the plane*/ -}}
    {{- $curAlias := dict "num" 1 -}}
    {{- range $alias := $plane.aliases | default list }}
-    {{- if $alias.pvc }}
+    {{- $aliasComponentMatches := or (not (hasKey $alias "components")) (has $component.name $alias.components) -}}
+    {{- if and $alias.pvc $aliasComponentMatches  }}
```
Comment (jakesmith) [inline helm/hpcc/templates/_helpers.tpl:360]: trivial/formatting: double space before }}

File: helm/hpcc/templates/_helpers.tpl [hunk 2]
```diff
@@ -390,11 +392,12 @@ The plane will generate a volume if it matches either an includeLabel or an incl
 {{- $planes := ($storage.planes | default list) -}}
 {{- $includeCategories := .includeCategories | default list -}}
 {{- $includeNames := .includeNames | default list -}}
+{{- $component := .me -}}
```
Comment (jakesmith) [inline helm/hpcc/templates/_helpers.tpl:395]: should update addVolumes helper description to say Pass in 'me' as well.

File: helm/hpcc/templates/_helpers.tpl [hunk 3]
```diff
@@ -417,7 +420,8 @@ The plane will generate a volume if it matches either an includeLabel or an incl
    {{- /*Generate entries for each alias of the plane*/ -}}
    {{- $curAlias := dict "num" 1 -}}
    {{- range $alias := $plane.aliases | default list }}
-    {{- if $alias.pvc -}}
+    {{- $aliasComponentMatches := or (not (hasKey $alias "components")) (has $component.name $alias.components) -}}
+    {{- if and $alias.pvc $aliasComponentMatches  }}
```
Comment (jakesmith) [inline helm/hpcc/templates/_helpers.tpl:424]: trivial/formatting: double space before }}

**Review by jakesmith — COMMENTED** [review]
@ghalliday - looks good, minor issues

File: helm/hpcc/templates/_helpers.tpl [hunk 4]
```diff
@@ -764,10 +768,12 @@ NB: uid=10000 and gid=10001 are the uid/gid of the hpcc user, built into platfor
 {{- $planes := ($storage.planes | default list) -}}
 {{- $includeCategories := .includeCategories | default list -}}
 {{- $includeNames := .includeNames | default list -}}
+{{- $component := .me -}}
 {{- range $plane := $planes -}}
  {{- if not $plane.disabled -}}
   {{- if and ($plane.forcePermissions) (or ($plane.pvc) (hasKey $plane "storageClass")) -}}
    {{- $mountpath := $plane.prefix -}}
+   {{- $componentMatches := or (not (hasKey $plane "components")) (has $component.name $plane.components) -}}
    {{- if or (has $plane.category $includeCategories) (has $plane.name $includeNames) -}}
```
[the remark this answers is not in the document]
Comment (ghalliday) [reply helm/hpcc/templates/_helpers.tpl]: good catch

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good to me.

**Pull request merged**